Handle missing $USER

In `share/rcm.sh.in` we say:

``` sh
        find "$hook_file" -type f \( \( -user $USER -perm -100 \) -o -perm -001 \)
```

But `$USER` is not guaranteed to be set. When unset, it leads to:

```
find: `-perm' is not the name of a known user
```

Fix this by either falling back to a known user (e.g. `${USER:-root}`) or checking for `$USER`'s value before passing it.

@mike-burns I'm not sure how to test this myself :), I tried to install this on a fresh ubuntu 16.04 docker container. Installed `build-essential` and `autotools`. I was able to run `./configure` but when I run `make` I'm getting this error:
```
make: *** No rule to make target 'man/rcm.7', needed by 'all-am'.  Stop.
```

--- a/share/rcm.sh.in
+++ b/share/rcm.sh.in
@@ -13,6 +13,10 @@ MKDIR=mkdir
 INSTALL=rcup
 ROOT_DIR="$HOME"
 
+if [ -z "$LOGNAME" ]; then
+  LOGNAME=$(whoami)
+fi
+
 ln_v() {
   $VERBOSE "'$1' -> '$2'"
   ln -s "$1" "$2"

--- a/test/rcup-hooks.t
+++ b/test/rcup-hooks.t
@@ -9,10 +9,10 @@ Pre-up and post-up hooks should run by default
   $ echo 'echo "example" > /tmp/test' > .dotfiles/hooks/pre-up
   > echo 'cat /tmp/test; rm /tmp/test' > .dotfiles/hooks/post-up
 
-  $ rcup
+  $ /usr/bin/env -i PATH="$PATH" HOME="$HOME" RCRC="$RCRC" RCM_LIB="$RCM_LIB" rcup
   example
 
 Ensure that hooks run when output of lsrc is non-empty
   $ touch .dotfiles/testrc
-  > rcup
+  $ /usr/bin/env -i PATH="$PATH" HOME="$HOME" RCRC="$RCRC" RCM_LIB="$RCM_LIB" rcup
   example

